src/main/scala/is/hail/stats/package.scala

```diff
@@ -299,6 +302,31 @@ package object stats {
     result.toInt
   }
 
+  def binomTest(nSuccess: Int, n: Int, p: Double, alternative: String): Double = {
+    val kind = alternative match {
+      case "two.sided" => TestKind.TWO_SIDED
+      case "less" => TestKind.LOWER
+      case "greater" => TestKind.GREATER
+      case _ => fatal("didn't recognize option for alternative. Use one of [less, greater, two.sided]")
+    }
+
+    DistributionTest.binomial_test(nSuccess, n, p, kind)(1)
+  }
+
+  def entropy[T](x: Traversable[T]): Double = {
+    val counts = x.foldLeft(mutable.Map[T, Double]())((counter, element) => {
+      counter.update(element, 1.0 + counter.getOrElse(element, 0.0))
+      counter
+    }).values
+
+    val length = counts.sum
+
+    -1.0 * counts.map(count => {
```

Comment:
no need to realize the intermediate array.  I'd accumulate by iterating in a while loop.

```
var entropy = 0.0
var i = 0
while (i < counts.length) {
  val prob = count / length
  entropy += prob / math.log(prob) 
}

entropy * math.log(2)
```
This exploits that log2(x) = log(x) / log(2) and avoids calling through all the layers of Breeze just to get the log2 function. I'll ask about stashing the value of log(2) somewhere in Hail instead.